kernel/src/engine/arrow_expression.rs

```diff
@@ -79,6 +79,14 @@ impl Scalar {
                 DataType::Map { .. } => unimplemented!(),
                 DataType::Struct { .. } => unimplemented!(),
```

Comment:
Seems like we need to implement this now, if we're going to support struct scalars?